specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json

```diff
@@ -0,0 +1,2640 @@
+          "readOnly": true
+        },
+        "channelSecret": {
+          "x-ms-secret": true,
+          "type": "string",
+          "description": "Secret for the line channel registration"
+        },
+        "channelAccessToken": {
+          "x-ms-secret": true,
+          "type": "string",
+          "description": "Access token for the line channel registration"
+        }
+      },
+      "description": "The properties corresponding to a line channel registration"
+    },
+    "DirectLineSpeechChannel": {
+      "type": "object",
+      "description": "DirectLine Speech channel definition",
+      "x-ms-discriminator-value": "DirectLineSpeechChannel",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Channel"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DirectLineSpeechChannelProperties",
+          "description": "The set of properties specific to DirectLine Speech channel resource"
+        }
+      }
+    },
+    "DirectLineSpeechChannelProperties": {
+      "properties": {
+        "cognitiveServicesSubscriptionId": {
+          "type": "string",
+          "description": "The cognitive service subscription ID to use with this channel registration."
+        },
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Whether this channel is enabled or not."
+        },
+        "customVoiceDeploymentId": {
+          "type": "string",
+          "description": "Custom speech model id (optional)."
+        },
+        "customSpeechModelId": {
+          "type": "string",
+          "description": "Custom voice deployment id (optional)."
+        },
+        "isDefaultBotForCogSvcAccount": {
+          "type": "boolean",
+          "description": "Make this a default bot for chosen cognitive service account."
+        }
+      },
+      "description": "The parameters to provide for the DirectLine Speech channel.",
+      "required": [
+        "cognitiveServicesSubscriptionId"
+      ]
+    },
+    "ChannelResponseList": {
+      "properties": {
+        "nextLink": {
+          "description": "The link used to get the next page of bot service channel resources.",
+          "type": "string"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BotChannel"
+          },
+          "description": "Gets the list of bot service channel results and their properties."
+        }
+      },
+      "description": "The list of bot service channel operation response."
+    },
+    "WebChatSite": {
+      "properties": {
+        "siteId": {
+          "type": "string",
+          "description": "Site Id",
+          "x-ms-mutability": [
+            "read"
+          ],
+          "readOnly": true
+        },
+        "siteName": {
+          "type": "string",
+          "description": "Site name"
+        },
+        "key": {
```

Comment:
Is this a new API?  It is generally a bad idea to have these proeprties null for some operations and provided in others.

Reply:
The *Channel definitions are here help sdk generation, notice the swagger itself doesn't refer to the specific channel definition.

Reply:
This is only acceptable if it would be a breaking change to remove these properties. This is an anti-pattern, as it breaks RPC requirements around the resource.  It is better to have the POST operation return a separate schema with keys only, rather than resource details with keys.

Reply:
To be clear, I am referring to the readOnly keys that only have values as the response in a POST operation

Reply:
I see, this is has not changed from previous version, see https://github.com/Azure/azure-rest-api-specs/blob/8e8cb6958f75504ec011a1e57f54b7e73d059531/specification/botservice/resource-manager/Microsoft.BotService/stable/2020-06-02/botservice.json#L1991